### DISCUSSION TITLE
Error when testing elastalert rules

### DISCUSSION THREAD

**Discussion by kennysden** [Security-Onion-Solutions/securityonion#2437] [Unsupported Versions] [answered]
I want to integrate slack with Security Onion, but still having trouble, I have updated to the lastest version 2.3.21. Why do I get this error to test elastalert rules?
![TestRules](https://user-images.githubusercontent.com/59517722/103000281-be498700-44f8-11eb-8720-10ad3ba63c28.png)

**Comment by weslambert** [comment, marked as answer]
Are you trying to test the all.rules file? That is a NIDS rule file.  Please try testing something in/from `/opt/so/rules/elastalert`.

**Reply by kennysden** [reply]
That was my first test, but this gives me the same error 
![TestRules2](https://user-images.githubusercontent.com/59517722/103001823-bdfebb00-44fb-11eb-975b-fae200e6b7a5.png)

**Reply by weslambert** [reply]
Okay, it's probably because the container doesn't know of that path.  Try substituting `/opt/elastalert/rules/` for `/opt/so/rules/elastalert`, so:

`/opt/elastalert/rules/$rulepath`

**Reply by kennysden** [reply]
So will I need to edit the script? Not sure where to make this change?

**Reply by weslambert** [reply]
No, I was merely referring the path you were specifying.  If you just run `so-elastalert-test`, IIRC, it should show you the available rules in the directory.  Just specify the path as it is displayed for the available rule(s).  This will be made clearer in the next release.
